### PR TITLE
HWP-271 Create test file for GET /api/community

### PR DESCRIPTION
**Pull request by mattsiel**
# Description

This PR includes the following proposed change(s):

- added a file to test the get community endpoint
- added tests related to authentication while using the get community request

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

**Comment by mattsiel** [comment]
added the start of the get community unit tests

File: app/server/tests/get_community.test.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  });
+
+  test('API returns description "Forbidden."', async () => {
+    let response = await request
+      .get('/community')
+      .set({authorization: `Bearer ${token}11`})
+
+    expect(response.text).toBe("Forbidden.");
+  });
+});
```
Comment (akroon3r) [inline app/server/tests/get_community.test.js:121]:
```suggestion
});

```

File: app/server/tests/get_community.test.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  test('API returns unsuccessful response', async () => {
+    let response = await request
+      .get('/community')
+      .set({authorization: `Bearer ${token}11`})
+
+    expect(response.ok).toBe(false);
+  });
+
+  test('API returns code 401', async () => {
```
Comment (mattsiel) [inline app/server/tests/get_community.test.js:106]:
```suggestion
  test('API returns code 403', async () => {
```

File: app/server/tests/get_community.test.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    let response = await request
+      .get('/community')
+      .set({authorization: `Bearer ${token}`})
+
+    expect(response.status).toBe(200);
+  });
+});
+
+
+describe('Testing the get communities function after loggin in, but without token', () => {
```
Comment (mattsiel) [inline app/server/tests/get_community.test.js:70]:
```suggestion
describe('Testing the get communities function after logging in, but without token', () => {
```

File: app/server/tests/get_community.test.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
+const endpoint = 'https://hwp-express-api-d63404-dev.apps.silver.devops.gov.bc.ca/api';
```
Comment (akroon3r) [inline app/server/tests/get_community.test.js:1]: We need to get this URL from an environment variable please. We cannot hard code URLs like this into our source.

Comment (akroon3r) [reply app/server/tests/get_community.test.js:1]: Also, file naming convention needs to follow the naming convention that Dylan, Brady, Zach and Brendan have set. I believe it's likeThis. Please confirm with the files that Dylan has already committed to the repo.

**Comment by BradyMitch** [comment]
Since every user is in the Welcome community upon registering, maybe check that the Welcome community is in the 200 response body?

File: app/server/tests/get_community.test.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      .get('/community')
+
+    expect(response.status).toBe(401);
+  });
+
+  test('API returns description "Not Authorized."', async () => {
+    let response = await request
+      .get('/community')
+
+    expect(response.text).toBe("Not Authorized.");
```
Comment (akroon3r) [inline app/server/tests/get_community.test.js:90]: Should this be Unauthorized?

**Review by akroon3r — COMMENTED** [review]
Good start, we need to address file naming convention and environment variables here before I can approve and merge please 😄

**Comment by akroon3r** [comment]
> Since every user is in the Welcome community upon registering, maybe check that the Welcome community is in the 200 response body?

Great suggestion.

File: app/server/tests/get_community.test.js (hunk 5 above)
Comment (mattsiel) [reply app/server/tests/get_community.test.js:90]: UnAuthorized is the error and Not Authorized is the error response I think

File: app/server/tests/get_community.test.js (hunk 4 above)
Comment (mattsiel) [reply app/server/tests/get_community.test.js:1]: will update the file name and try to get the environment variables 👍

Comment (BradyMitch) [reply app/server/tests/get_community.test.js:1]: yes we are using camelCase

Comment (mattsiel) [inline app/server/tests/get_community.test.js:1]:
```suggestion
const endpoint = `http://${window._env_.REACT_APP_API_REF}:${window._env_.REACT_APP_API_PORT}/api`;
```

Comment (BradyMitch) [reply app/server/tests/get_community.test.js]: Is this test meant to use the openshift url or local dev? The way you are using the env variables her is for local dev.

File: app/server/tests/getCommunity.test.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
+const endpoint = `http://${window._env_.REACT_APP_API_REF}:${window._env_.REACT_APP_API_PORT}/api`;
```
Comment (mattsiel) [inline app/server/tests/getCommunity.test.js:1]:
```suggestion
const endpoint = `http://${window._env_.REACT_APP_API_REF}/api`;
```

File: app/server/tests/getCommunity.test.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
+const endpoint = `http://${window._env_.REACT_APP_API_REF}/api`;
```
Comment (mattsiel) [inline app/server/tests/getCommunity.test.js:1]:
```suggestion
const endpoint = `http://${window._env_.REACT_APP_API_REF}`;
```

**Pull request closed without merging**